### PR TITLE
Fix ordered_dict.h for CUDA on Windows

### LINKED ISSUE #55266: Building with MSVC 2019 Community Edition fails
## 🐛 Bug

Build fails on Windows with latest stable version of everything:

function "torch::OrderedDict<Key, Value>::Item::operator=(const torch::OrderedDict<std::string, at::Tensor>::Item &) [with Key=std::string, Value=at::Tensor]" (declared implicitly) cannot be referenced -- it is a deleted function

## To Reproduce

Trying to build this repo ( https://github.com/sunwj/CAR ) with MSVC 2019 Community Edition and latest stable PyTorch fails:

C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/include\xutility(4643): error: function "torch::OrderedDict<Key, Value>::Item::operator=(const torch::OrderedDict<std::string, at::Tensor>::Item &) [with Key=std::string, Value=at::Tensor]" (declared implicitly) cannot be referenced -- it is a deleted function
detected during:
instantiation of "_OutIt std::_Move_unchecked1(_InIt, _InIt, _OutIt, std::false_type) [with _InIt=torch::OrderedDict<std::string, at::Tensor>::Item *, _OutIt=torch::OrderedDict<std::string, at::Tensor>::Item *]"
(4659): here
instantiation of "_OutIt std::_Move_unchecked(_InIt, _InIt, _OutIt) [with _InIt=torch::OrderedDict<std::string, at::Tensor>::Item *, _OutIt=torch::OrderedDict<std::string, at::Tensor>::Item *]"
C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/include\vector(1353): here
instantiation of "std::vector<_Ty, _Alloc>::iterator std::vector<_Ty, _Alloc>::erase(std::vector<_Ty, _Alloc>::const_iterator) [with _Ty=torch::OrderedDict<std::string, at::Tensor>::Item, _Alloc=std::allocator<torch::OrderedDict<std::string, at::Tensor>::Item>]"
C:/Users/leon/AppData/Local/Programs/Python/Python39/lib/site-packages/torch/include\torch/csrc/api/include/torch/ordered_dict.h(419): here
instantiation of "void torch::OrderedDict<Key, Value>::erase(const Key &) [with Key=std::string, Value=at::Tensor]"
C:/Users/leon/AppData/Local/Programs/Python/Python39/lib/site-packages/torch/include/torch/csrc/api/include\torch/nn/modules/container/parameterdict.h(51): here

## Expected behavior

Build fails after running:
cd adaptive_gridsampler
python setup.py build_ext --inplace

## Environment

 - PyTorch Version Stable 1.8.1
 - OS: Windows
 - How you installed PyTorch: pip install torch==1.8.1+cu102 torchvision==0.9.1+cu102 torchaudio===0.8.1 -f https://download.pytorch.org/whl/torch_stable.html
 - Python version: Python 3.9
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: RTX 2080 Ti

cc @malfet @seemethere @walterddr @peterjc123 @maxluk @nbcsm @guyang3532 @gunandrose4u @mszhanyi @skyline75489

### PR DESCRIPTION
**Pull request by peterjc123**
Fixes https://github.com/pytorch/pytorch/issues/55266

### CODE REVIEW

File: torch/csrc/api/include/torch/ordered_dict.h [hunk 1]
```diff
@@ -193,6 +193,16 @@ class OrderedDict<Key, Value>::Item {
   /// Constructs a new item.
   Item(Key key, Value value) : pair_(std::move(key), std::move(value)) {}
 
+#if defined(__CUDACC__) && (__CUDACC_VER_MAJOR__ < 11) && defined(_MSC_VER)
```
Comment (skyline75489) [inline torch/csrc/api/include/torch/ordered_dict.h:196]: Excuse my innocence but I don’t quite get it why CUDA version matters here?

Comment (peterjc123) [reply torch/csrc/api/include/torch/ordered_dict.h:196]: You may know more context here. https://github.com/pytorch/pytorch/pull/47524#issuecomment-736018820

**Comment by peterjc123** [comment]
@ezyang @malfet Windows tests (CUDA 10.1) have passed. Please take a look.

**Review by ezyang — APPROVED** [review]
okey dokey

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**